### PR TITLE
Add 'rebootable' tag to all nodes except 'unique'

### PR DESCRIPTION
**Pull request by dmick**
This corresponds to a change in the ceph-build-pull-requests
job to request a rebootable node

Signed-off-by: Dan Mick <dan.mick@redhat.com>

### CODE REVIEW

**Comment by dmick** [comment]
https://github.com/ceph/ceph-build/pull/356 depends on this

File: deploy/playbooks/roles/common/templates/prod_nodes.py.j2 [hunk 1]
```diff
@@ -10,7 +10,7 @@ nodes = {
         'keyname': keyname,
         'image_name': 'Ubuntu 12.04',
         'size': 'vps-ssd-1',
-        'labels': ['amd64', 'x86_64', 'precise', 'small'],
+        'labels': ['amd64', 'x86_64', 'precise', 'small', 'rebootable'],
```
Comment (andrewschoen) [inline deploy/playbooks/roles/common/templates/prod_nodes.py.j2]: You'll need to add the new label in the curl call to prado as well. That's how prado knows how what labels to use when registering the new slave in jenkins.

Comment (dmick) [inline deploy/playbooks/roles/common/templates/prod_nodes.py.j2]: thanks, looking into it

Comment (dmick) [inline deploy/playbooks/roles/common/templates/prod_nodes.py.j2]: Added another commit

**Comment by alfredodeza** [comment]
I might be missing a bit from the discussion yesterday, but I thought the idea of the new `rebootable` label was to allow the ceph-pull-requests job have a better selection of the nodes preventing a situation where a node with a matching subset of labels would be servicing the job.

However, this label is being applied to every node except for the ones with the `unique` label. Can you explain why this is needed on all those nodes?

**Comment by alfredodeza** [comment]
I think that rather than changing all the other nodes, we should add two nodes that have and service the `bootable` label. For example Centos7 would be:

    'centos7_huge_rebootable': {
        'script': dedent("""#!/bin/bash
        curl -u {{ prado_user | default('admin') }}:{{ prado_token }} -L "{{ prado_url }}/setup/slave/?token={{ jenkins_prado_token }}&executors=2&labels=rebootable+amd64+centos7+huge+x86_64&nodename=centos7_huge_rebootable__%s" | bash"""),
        'keyname': keyname,
        'image_name': 'Centos 7',
        'size': 'hg-30',
        'labels': ['amd64', 'x86_64', 'centos7', 'huge', 'rebootable'],
        'provider': 'openstack'
    },

And we would add another one for trusty "huge". That would add two more configs and leave everything else as-is

**Comment by dmick** [comment]
All the nodes marked are rebootable. It's not aimed at one particular job; it's designed to distinguish mita nodes from physical nodes.

Other tags further qualify the nodes for ceph-p-r.

Sent from Nine

**Comment by alfredodeza** [comment]
Adding `rebootable` to nodes to distinguish them as "mita nodes" doesn't read right, since it is leaving out nodes that are labeled as "unique". Those nodes are also mita nodes.

If nodes need to have some identifier to differentiate them from "created by mita" vs. "other nodes" on a label-basis I think we should have a label that describes just that

**Comment by andrewschoen** [comment]
Is the main goal here to keep the arm64 nodes from being selected by ceph-pull-requests and not risk the chance of being rebooted?  Perhaps just adding ``x86_64`` to ``nodes`` on that job would solve that.

https://github.com/ceph/ceph-build/blob/master/ceph-pull-requests/config/definitions/ceph-pull-requests.yml#L6

So it would become: ``huge && x86_64 && (centos7 || trusty)``

**Comment by dmick** [comment]
No,the label describes exactly the attribute of the nodes that matters. I left out the unique nodes because unique is already muddy as a tag. 

Sent from Nine

**Comment by dmick** [comment]
No,the goal is to describe the attribute of the node that makes it different and desirable.  That's only true today for one job, but the point is to be clear and explicit.

Sent from Nine

**Comment by alfredodeza** [comment]
I am not disagreeing in adding more labels. The more labels and the more they can describe attributes of a node the better.

We shouldn't drag the discussion around the `unique` label because that was a thing that we tried to implement quickly out of despair (it needs to be revisited for sure).

Because we are adding a new architecture (arm64/aarch64) then I think that it would be best to declare on other jobs, explicitly, "x86_64" or "amd64" (they are both present currently on every node).

It is reasonable for other jobs now to be burdened with having to "ask for the explicit arch it needs".

Would you mind adding those to this PR aside from the `rebootable` label? 

Also minor thing here: this repo should probably be moved, not forked. Mita/prado/chacra where all born to make CI awesome again ™ for Ceph :)

**Comment by dmick** [comment]
Well, I think the jobs that *require* x86_64 already ask for it, but I'll review.  Part of the point here is that especially if we're dedicating physical ARM hosts as slaves, there's no reason they can't run architecture-independent or distro-independent jobs too.

As for moving, sure, fine with me, I just wanted a place I could submit a PR.

**Comment by alfredodeza** [comment]
Can you open this PR against the mita at my account so I can merge there and then move here?

**Comment by dmick** [comment]
Closed in favor of https://github.com/alfredodeza/mita/pull/54

**Pull request closed without merging**